### PR TITLE
Update swagger - including other status codes, better sharing of stuffs

### PR DESCRIPTION
**Pull request by msluther**
Some basic swagger cleanup.
cleanup

Adding errors codes.
404 surface not found

### CODE REVIEW

File: lib/routes/swagger.js [hunk 1]
```diff
@@ -43,6 +43,71 @@ module.exports = function (app) {
...
+   *       description: OK
+   *     400:
+   *       $ref: '#/responses/Standard400'
+   *     403:
+   *       $ref: '#/responses/Standard403'
+   *     404:
+   *       $ref: '#/responses/Standard404'
+   *     500:
+   *       $ref: '#/responses/Standard500'
+   *   Standard304:
```
Comment (jcrugzz) [inline lib/routes/swagger.js]: this should be 204 right?

File: lib/npm/routes.js [hunk 2]
```diff
@@ -59,6 +53,14 @@ module.exports = function (app) {
    * definitions:
    *   DistTags:
    *     type: object
+   * parameters:
```
Comment (DanielSanudo) [inline lib/npm/routes.js]: Why can't be this with the others in `lib/routes/swagger.js`?

File: lib/routes/swagger.js (hunk 1 above)
Comment (msluther) [reply lib/routes/swagger.js]: 304. https://github.com/godaddy/warehouse.ai/blob/master/lib/npm/routes.js#L272

File: lib/npm/routes.js (hunk 2 above)
Comment (msluther) [reply lib/npm/routes.js]: The general strategy I was trying to follow here was:

If it's used by a single route, just document it there,
If it's used by a single file, put the shared definition in that file,
If it's used by more than one file, put the shared definition in `lib/routes/swagger.js`

Could change that and move all the shared definitions into swagger.js I'm not sure I feel too strongly about it

File: lib/routes/swagger.js (hunk 1 above)
Comment (msluther) [reply lib/routes/swagger.js]: There are some 204s but I left them in place. I could have doc'd the 304 in place as well and probably should have.

Comment (jcrugzz) [reply lib/routes/swagger.js]: @msluther I was being dumb and forgot 304 is for the caching bit and not an awkward redirect code

File: lib/npm/routes.js (hunk 2 above)
Comment (DanielSanudo) [reply lib/npm/routes.js]: I would prefer something like:
If it's used by a single route, just document it there,
If it's used by more than once then in the shared location.

I was just curious, not a blocking / strong point.

**Pull request merged**